gtsfm/scene_optimizer.py

```diff
@@ -274,13 +277,27 @@ def _filter_dict_keys(dict: Dict[Any, Any], ref_dict: Dict[Any, Any]) -> Dict[An
         if self._save_gtsfm_data:
             auxiliary_graph_list.extend(save_gtsfm_data(image_graph, ba_input_graph, ba_output_graph))
 
+        img_dict_graph = dask.delayed(get_image_dictionary)(image_graph)
+        dense_points_graph = self.dense_multiview_optimizer.create_computation_graph(img_dict_graph, ba_output_graph)
+        mvs_ply_save_fpath = str(RESULTS_PATH / "mvs_output" / "dense_pointcloud.ply")
+        auxiliary_graph_list.append(
+            dask.delayed(io_utils.save_point_cloud_as_ply)(save_fpath=mvs_ply_save_fpath, points=dense_points_graph)
+        )
+
         # as visualization tasks are not to be provided to the user, we create a
         # dummy computation of concatenating viz tasks with the output graph,
         # forcing computation of viz tasks
         output_graph = dask.delayed(lambda x, y: (x, y))(ba_output_graph, auxiliary_graph_list)
+        ba_output_graph = output_graph[0]
 
         # return the entry with just the sfm result
-        return output_graph[0]
+        return ba_output_graph
+
+
+def get_image_dictionary(image_list: List[Image]) -> Dict[int, Image]:
```

Comment:
in a future PR, I proposed we pass the loader directly into the SceneOptimizer's create_computational_graph(). Out of scope for this PR though.

Reply:
Thanks for the review Ayush.

I am seeing a bug slowdown on Skydio 501, and I think it may be due to RAM usage. By passing in the images graph instead of the loader, I believe we load every image into memory at some points. I think this could be avoided by passing in the loader. I still need to verify this, though.

Reply:
Oh I see. So you propose having multiple nodes to get images, so that we prevent node reuse which keeps it in memory.

I think its a good idea, and lets do it for the whole repo at once. Given we have learnt a few things about dask, it could be a good time to rethink the graph and API design.